### PR TITLE
[bct] Clean up async reporting

### PR DESCRIPTION
**Pull request by catalinaperalta**
Clean up duplicate sync and async changes. 

Fixes https://github.com/Azure/azure-sdk-for-python/issues/36973
Part of https://github.com/Azure/azure-sdk-for-python/issues/36763

### CODE REVIEW

File: scripts/breaking_changes_checker/changelog_tracker.py [hunk 1]
```diff
@@ -46,6 +46,19 @@ def __init__(self, stable: Dict, current: Dict, diff: Dict, package_name: str, *
...
+    def run_async_changelog_cleanup(self) -> None:
+        # Create a list of all sync changes
+        non_aio_changes = [fa for fa in self.features_added if "aio" not in fa[2]]
+        # Remove any aio change if there is a sync change that is the same
+        for change in non_aio_changes:
+            for c in self.features_added:
+                if "aio" in c[2]:
+                    if change[1] == c[1] and change[3:] == c[3:]:
+                        self.features_added.remove(c)
+                        break
```
Comment (msyyc) [inline scripts/breaking_changes_checker/changelog_tracker.py:61]: It seems duplicated code and it is better to extract a separated function.

Comment (catalinaperalta) [reply scripts/breaking_changes_checker/changelog_tracker.py]: Done,thanks!

**Review by msyyc — APPROVED** [review]
I test the PR locally with real service and it indeed remove duplicated items from async api

**Pull request merged**